csrc/scheduler/vectorize_helper.cpp

```diff
@@ -842,6 +842,84 @@ std::vector<std::unordered_map<TensorView*, Val*>> getTvToContigInnerSizeMapsOf(
   return mappers;
 }
 
+// The returns a map of input to their resize alignment info. This function
+// currently checks the alignment requirement on contiguous iterdomain in
+// allocation domain that *could be* broken by resize. This function returns a
+// map for TensorView, containing the index in their allocation domain whose
+// stride needs to be checked in alignment. e.g. For TensorView with size {4,
+// 1025}, assuming it's contiguous on all dimensions. If we slice the second
+// dimension with slice(0, 1024, 1). The output tensor would have shape {4,
+// 1024}. The slice prevents us from collapsing the indexing on its outer
+// dimension. At this point, due to the resize operation on dimension-1024, we
+// now need to check the stride on dimension-4 for alignment. NOTE: This is a
+// conservative check, not all resize breaks the contiguity of its inner
+// dimension, i.e. think about a combination of pad/slice that cancels out each
+// other;
+std::unordered_map<TensorView*, TensorResizeAlignmentInfo>
+mapResizeAlignmentToInputs(TensorView* ref) {
+  std::unordered_map<TensorView*, TensorResizeAlignmentInfo> res;
+  auto id_model =
+      std::make_unique<IdModel>(ref->fusion(), /*build_graphs=*/false);
+  id_model->buildExactGraph();
+  auto& exact_graph = id_model->idGraph(IdMappingMode::EXACT);
+
+  // ref_target_domains is the scheduling tensor
+  ValGroups ref_target_domains = exact_graph.toGroups(ref->getLogicalDomain());
+
+  // propagate from input to ref_target_domains
+  auto in_tvs = ir_utils::filterByType<TensorView>(ref->fusion()->inputs());
```

Comment:
Do we need to check all inputs? Didn't we have a list of vectorized inputs?

Reply:
No we don't.

This one finds the map for all inputs. When we later check for alignment requirements for the list of vectorized inputs, we find the candidate in the map.

We can use the list of vectorized inputs/outputs instead of only looking at the inputs to have a smaller map. But functionally they are about the same.